src/System/Router/interceptLinks.ts

```diff
@@ -12,9 +12,10 @@ interface InterceptLinksProps {
  */
 export function interceptLinks({ router, routes }: InterceptLinksProps) {
   catchLinks(window, href => {
+    const [pathname, _search] = href.split("?")
     // FIXME: PR upstream; `matchRoutes` is missing from type definition
     // @ts-ignore
-    const foundUrl = router.matcher.matchRoutes(routes, href)
+    const foundUrl = router.matcher.matchRoutes(routes, pathname)
```

Comment:
Finally got around to fixing this long-standing issue. 

Before, what would happen is that our `<RouterLink to='/some-path?foo=bar'>` link would perform hard jumps between pages because the additional query param wouldn't match against our routes, and return false (we don't abstractly define query params in our routes definitions). And so this fixes that by splitting on the query param and matching _only_ against the pathname. Quick and easy